### PR TITLE
Add test for New-Item -Force

### PR DESCRIPTION
**Pull request by robdy**
# PR Summary

I created new test for checking if `New-Item -Force` creates a file if the folder in path is missing. It also checks that when `-Force` is not used, the command fails.

## PR Context

This is my first PR so feel free to give your feedback what I could do better :)

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 1]
```diff
@@ -294,3 +294,27 @@ Describe "New-Item with links fails for non elevated user if developer mode not
...
+        { New-Item $FullyQualifiedFile -ErrorAction Stop } | Should -Throw -ErrorId 'NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand'
+    }
+	It "Should create new file correctly when -Force is used and folder in the path doesn't exist" {
+	    # Explicitly removing folder first
+	    Remove-Item $FullyQualifiedFolder -ErrorAction SilentlyContinue
+	    Test-Path -Path $FullyQualifiedFolder | Should -BeFalse
+		
+        { New-Item $FullyQualifiedFile -Force -ErrorAction Stop } | Should -Not -Throw
+    }
+}
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Add an end of line here.

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 2]
```diff
@@ -294,3 +294,27 @@ Describe "New-Item with links fails for non elevated user if developer mode not
...
+	    Test-Path -Path $FullyQualifiedFolder | Should -BeFalse
+		
+        { New-Item $FullyQualifiedFile -ErrorAction Stop } | Should -Throw -ErrorId 'NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand'
+    }
+	It "Should create new file correctly when -Force is used and folder in the path doesn't exist" {
+	    # Explicitly removing folder first
+	    Remove-Item $FullyQualifiedFolder -ErrorAction SilentlyContinue
+	    Test-Path -Path $FullyQualifiedFolder | Should -BeFalse
+		
+        { New-Item $FullyQualifiedFile -Force -ErrorAction Stop } | Should -Not -Throw
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Please all add verification for whether the file was created. 

```PowerShell
$FullyQualifiedFile | Should -Exist 
```

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@PoshChan Please remind me in 24 hours

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 3]
```diff
@@ -294,3 +294,30 @@ Describe "New-Item with links fails for non elevated user if developer mode not
         $TestFilePath | Should -Exist
     }
 }
+
+Describe "New-Item -Force allows to create an item even if the directories in the path don't exist" -Tags "CI" {
+    BeforeAll {
+        $testFile             = 'testfile.txt'
+        $testFolder           = 'testfolder'
+		$FullyQualifiedFolder = Join-Path -Path $TestDrive -ChildPath $testFolder
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Indentation. Please use spaces.

Please check indentations below too.

Comment (robdy) [reply test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Good catch, fixed!

**Review by iSazonov — APPROVED** [review]
@robdy Thanks for your contribution!

**Comment by PoshChan** [comment]
@daxian-dbw, this is the reminder you requested 24 hours ago

**Pull request merged**